src/muc_light/mod_muc_light_db_rdbms.erl

```diff
@@ -615,17 +615,13 @@ force_clear() ->
 create_room_transaction(MainHost, {RoomU, RoomS}, Config, AffUsers, Version) ->
     insert_room(MainHost, RoomU, RoomS, Version),
     RoomID = mongoose_rdbms:selected_to_integer(select_room_id(MainHost, RoomU, RoomS)),
-    lists:foreach(
-      fun({{UserU, UserS}, Aff}) ->
-              {updated, _} = insert_aff(MainHost, RoomID, UserU, UserS, Aff)
-      end, AffUsers),
-    ConfigFields = mod_muc_light_room_config:to_binary_kv(
-                     Config, mod_muc_light:config_schema(RoomS)),
-    lists:foreach(
-      fun({Key, Val}) ->
-              {updated, _} = insert_config(MainHost, RoomID, Key, Val)
-      end, ConfigFields),
-      ok.
+    Schema = mod_muc_light:config_schema(RoomS),
+    ConfigFields = mod_muc_light_room_config:to_binary_kv(Config, Schema),
+    [insert_aff(MainHost, RoomID, UserU, UserS, Aff)
+     || {{UserU, UserS}, Aff} <- AffUsers],
+    [insert_config(MainHost, RoomID, Key, Val)
+     || {Key, Val} <- ConfigFields],
```

Comment:
I liked the previous version better as this one silently ignores wrong values, e.g. if there is a `{kitty, purr}` in `AffUsers`.